### DISCUSSION TITLE
Implicit many-to-many relations only index one foreign key

### DISCUSSION THREAD

**Discussion by JSRossiter** [prisma/prisma#13755] [Q&A]
Why does only the B field have an index created for relation tables? I would expect this relation to be bi-directional and indexes have a large performance impact on joins, which prisma uses when you query based on the presence of a relation.

Is it possible to define the relation table in the schema (or otherwise add the missing index) while maintaining the code benefits of implicit relation? If not I'll likely add it direct to the db, but that obviously isn't ideal.

https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations

**Comment by janpio** [comment]
There is a unique index on `A,B`, and an additional index on `B`:
```sql
CREATE TABLE "Category" (
    id SERIAL PRIMARY KEY
);
CREATE TABLE "Post" (
    id SERIAL PRIMARY KEY
);
-- Relation table + indexes -------------------------------------------------------
CREATE TABLE "_CategoryToPost" (
    "A" integer NOT NULL REFERENCES "Category"(id),
    "B" integer NOT NULL REFERENCES "Post"(id)
);
CREATE UNIQUE INDEX "_CategoryToPost_AB_unique" ON "_CategoryToPost"("A" int4_ops,"B" int4_ops);
CREATE INDEX "_CategoryToPost_B_index" ON "_CategoryToPost"("B" int4_ops);
```

That should cover all joins you can do with that table (as the compound unique will work the same as an additional index on `A` - or are you observing something else?).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
